Updates Netlify Adapter docs for remote images (#7375)
withastro/docs: src/content/docs/en/guides/integrations-guide/netlify.mdx

```js title="astro.config.mjs" ins={8}
import { defineConfig } from 'astro/config';

> import netlify from '@astrojs/netlify';

Reviewer: @ascorbic why the update of the import?

```js title="astro.config.mjs" ins={8}
import { defineConfig } from 'astro/config';

> import netlify from '@astrojs/netlify';

Reviewer: @ascorbic why the update of the import?

Author: The code sample was wrong. The import changed in a previous version.

Author: See https://github.com/withastro/adapters/blob/main/packages/netlify/src/functions.ts

});
```

> If you are using images hosted on another domain, you must authorize the domain or URL patterns using the [`image.domains`](/en/reference/configuration-reference/#imagedomains) and [`image.remotePatterns`](/en/reference/configuration-reference/#imageremotepatterns) options:

Reviewer 2: ```suggestion
If you are using images hosted on another domain, you must authorize the domain or URL patterns using the [`image.domains`](/en/reference/configuration-reference/#imagedomains) or [`image.remotePatterns`](/en/reference/configuration-reference/#imageremotepatterns) configuration options:
```
"and" sounds like they may have to configure both!

```js title="astro.config.mjs" ins={8-10}
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
    // ...
    output: 'server',
    adapter: netlify(),
    image: {
      domains: ['example.com'],
    },
});
```
For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images/). This is not required for images hosted on the same domain as your site.

> ### Static sites & Redirects

Reviewer 2: ```suggestion
### Static sites with the Netlify Adapter
```
Now that this section is about MORE than redirects, what do we think about generalizing the section?

> For static sites you usually don't need an adapter unless you are using the image service. However, if you use `redirects` configuration in your Astro config, the Netlify adapter can be used to translate this to the proper `_redirects` format.

Reviewer 2: ```suggestion
For static sites (`output: 'static'`) hosted on Netlify, you usually don't need an adapter. However, some deployment features are only available through an adapter.

Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-mage-cdn-support).

If you use `redirects` configuration in your Astro config, the Netlify adapter can be used to translate this to the proper `_redirects` format.
```
Make this accurate, but I think this better sets up this section for "static sites that need to use this adapter"

Author: Yeah, that's definitely better

});
```

If you are using images hosted on another domain, you must authorize the domain or URL patterns using the [`image.domains`](/en/reference/configuration-reference/#imagedomains) or [`image.remotePatterns`](/en/reference/configuration-reference/#imageremotepatterns) configuration options:

```js title="astro.config.mjs" ins={8-10}
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
    // ...
    output: 'server',
    adapter: netlify(),
    image: {
      domains: ['example.com'],
    },
});
```

> For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images/). This is not required for images hosted on the same domain as your site.

Reviewer: ```suggestion
For more information, see [the guide to authorizing remote images](/en/guides/images/#authorizing-remote-images). This is not required for images hosted on the same domain as your site.
```

### Static sites with the Netlify Adapter

For static sites (`output: 'static'`) hosted on Netlify, you usually don't need an adapter. However, some deployment features are only available through an adapter.

> Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-mage-cdn-support).

Reviewer: ```suggestion
Static sites will need to install this adapter to use and configure Netlify's [image service](#netlify-image-cdn-support).
```
